### ISSUE TITLE
Easy way to unhide questions?

### ISSUE THREAD

**Issue by kaushalmodi** [vermiculus/sx.el#281]

**Comment by Malabarba** [comment]
Good point. What interface could we use for unhiding a question? @vermiculus

**Comment by kaushalmodi** [comment]
Can you use `completing-read` to display the list of already hidden questions? When user hits `RET` on the selected question, that one gets unhidden. 

- It will be useful to store the question string in the `hidden-questions.el` too so that we have something to display in the `completing-read` list instead of just the question numbers.

- I don't know how we can enable multiple selection using this interface though if that's a feature planned to be added later.

**Comment by vermiculus** [comment]
I think using `completing-read` is be a good idea, but it require storing extra information in the file (which I'd really like to avoid, personally -- this file can blow up).

The 'most pure' option is to allow popping a question off this list and re-inserting it into the question list data structure we have. This would effectively 'undo' the last hide, but having the ability to pick-and-choose which question you want to unhide is very appealing.

In addition to the simple pop/repaint solution, there might be room for hiding/unhiding a question from the question/answer view. This might be useful if we ever decide to maintain an organized index of questions -- but that time has not yet come :)

**Comment by kaushalmodi** [comment]
How about moving the hidden questions to an ''sx-hidden-tab' tab. When user
hits the 'h' key on a question in that tab, the question becomes unhidden
once again and shows up when the user switches to the All Questions tab.

**Comment by Malabarba** [comment]
The hidden-tab suggestion would be awesome, but I personally won't have the time to do that. =/
If anyone would like to try, it's actually quite easy to define a new tab (see the `sx-tab` file) and the `sx-question-list-mode` (which is used by the tabs) is very well documented.

`completing-read` is trivial to implement, but asking for the quesiton number is sort of irrelevant (and I don't think we should store strings either).

Finally, poping a question off the hidden-list is easy too. The only issue is that they are stored in the list ordered by ID, not by most-recently-hidden. So the user might have to pop quite a few in order to get back the latest one he hid.
